overlord/snapstate/snapmgr.go

```diff
@@ -147,6 +154,46 @@ func (snapst *SnapState) CurrentSideInfo() *snap.SideInfo {
 	panic("cannot find snapst.Current in the snapst.Sequence")
 }
 
+func (snapst *SnapState) PreviousSideInfo() *snap.SideInfo {
+	n := len(snapst.Sequence)
+	if n < 2 {
+		return nil
+	}
+	// find "current" and return the one before that
+	currentIndex := snapst.currentIndex()
+	if currentIndex == 0 {
+		return nil
+	}
+	return snapst.Sequence[currentIndex-1]
+}
+
+func (snapst *SnapState) findIndex(rev snap.Revision) int {
+	for i, si := range snapst.Sequence {
+		if si.Revision == rev {
+			return i
+		}
+	}
+	return -1
+}
+
+func (snapst *SnapState) currentIndex() int {
```

Comment:
Doesn't seem worth its weight. The one-liner below seems okay to use for those cases too.